ccswbs/ugnext
# Add Google Tag Manager code to buttons

# Summary of changes

Add Google Tag Manager click-tracking code to buttons

## Frontend
- Add `onClick` prop to `button.js`
- Add analytics variables and `onClick` function to `button-section.js`
- Add analytics schema to `widgets.graphql`

### Incidental changes
- Fix undefined section column name error in `section-widget.js` through proper null check
- Cherry-pick previous fix for "No Menu" value
- Add additional environment variable for dev version of GTM ID

## Backend

N/A

# Test plan
- Go to [Google Tag Manager](https://tagmanager.google.com) and log in 
- Run [Preview](https://support.google.com/tagmanager/answer/6107056?hl=en).
- Paste the URL https://deploy-preview-25--ugnext.netlify.app/widget-examples which will then open in a separate tab
- Return to the Tag Assistant tab and click the `Data Layer` option
- Go to https://deploy-preview-25--ugnext.netlify.app/widget-examples#buttons and click the first CTA button
- Switch back to the Tag Assistant tab and verify the custom event details show up
